src/rdkafka_cgrp.c

```diff
@@ -789,6 +1100,134 @@ rd_kafka_cgrp_assignor_run (rd_kafka_cgrp_t *rkcg,
                      rkas->rkas_protocol_name->str,
                      member_cnt);
 
+        /* https://cwiki.apache.org/confluence/display/KAFKA/KIP-429%3A+Kafk\
+           a+Consumer+Incremental+Rebalance+Protocol */
+        if (rkas->rkas_supported_protocols ==
+            RD_KAFKA_REBALANCE_PROTOCOL_COOPERATIVE) {  // TODO: Support assignors with multiple protocols.
+                int i;
```

Comment:
ok, extracted. i left the unit testing as a maybe-todo, i think i'd need to factor out the rkcg. this function will be well tested by integration tests, though unit testing would allow greater coverage more easily.